### PR TITLE
Flutter upgraded to 3.3.8 - Issue #584

### LINKED ISSUE #584: Upgrade Flutter
**Is your feature request related to a problem? Please describe.**
There are deprecated functions being used in the application. This prevents upgrading to the new flutter version. 

**Describe the solution you'd like**
Fix the application so the flutter version can be upgraded

### PR DESCRIPTION
**Pull request by andreaslborg**
# Description
Flutter is now upgraded to 3.3.8.
A lot of changes have been made in order to upgrade flutter. As seen below, 104 files needed an update as many things were deprecated.

This pull request is closely related to [this PR](https://github.com/aau-giraf/api_client/pull/122) on api_client. 

Fixes #584

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- Runs without errors on the weekplanner application
- Tested on physical tablet and emulator
- Tested with `flutter test --coverage`
- 0 errors, 0 warnings and 0 hints were given by dart analysis. This can be checked with `flutter analyze`

**Development Configuration**
*Type "flutter --version" and "dart --version" in your CMD to check versions.*

* Flutter version: 3.3.8
* Dart version: 2.18.4
* IDE: Android Studio 2021.3.1 Patch 1

## UPGRADE GUIDE

1. Upgrade Flutter in cmd (this will give you the latest version, which is 3.3.8) or download the specific version [here](https://docs.flutter.dev/development/tools/sdk/releases) and replace your old Flutter folder.
`flutter upgrade`

2. Update dependencies in path ..._\weekplanner_ :
`flutter pub upgrade`

3. Run flutter doctor and analyze to check for issues:
`flutter doctor`
`flutter analyze`

4. Lastly, if you experience issues with JAVA, we recommend that you update your IDE. We used Android Studio Dolphin (2021.3.1), which can be found here: https://developer.android.com/studio/archive

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works, if necessary
- [X] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [X] I have Acceptance Tested this on an Android device

### CODE REVIEW

**Comment by andreaslborg** [comment]
This PR is not ready for checks before https://github.com/aau-giraf/api_client/pull/122 is merged.

File: lib/main.dart [hunk 1]
```diff
@@ -3,26 +3,26 @@ import 'package:flutter/material.dart';
...
+  Bootstrap().register();
   WidgetsFlutterBinding.ensureInitialized();
 
   /***
    * The weekplanner will by default run towards the dev-enviroment
    * Use the "environments.local.json" for running against your local web-api
    * For IOS users: change the SERVER_HOST in the environment.local file to "http://localhost:5000"
    */
-  environment.setFile('assets/environments.dev.json').whenComplete(() {
+  environment.setFile('assets/environments.local.json').whenComplete(() {
```
Comment (MarcMData) [inline lib/main.dart]: I'm assuming this should be set to the dev environment? If not, then the comment above it should be changed

**Review by MarcMData — COMMENTED** [review]
Tested locally on a Pixel C API 31 emulator.  Upgraded flutter as per the guide in the PR without issue. Had to install additional dependencies in Visual Studio, as advised from running Flutter Doctor.

Everything works as it should. I do get one warning: 
"W/FlutterActivityAndFragmentDelegate( 4241): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps." 
Not sure what that is referencing, but it doesn't seem to be interfering with any functionality.

File: lib/main.dart (hunk 1 above)
Comment (andreaslborg) [reply lib/main.dart]: It should be "dev" - I've updated main.dart

**Comment by andreaslborg** [comment]
> Everything works as it should. I do get one warning: "W/FlutterActivityAndFragmentDelegate( 4241): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps." Not sure what that is referencing, but it doesn't seem to be interfering with any functionality.

Thanks for pointing it out. We will have another look at this.

**Comment by Bisgaard01** [comment]
> Everything works as it should. I do get one warning: "W/FlutterActivityAndFragmentDelegate( 4241): A splash screen was provided to Flutter, but this is deprecated. See flutter.dev/go/android-splash-migration for migration steps." Not sure what that is referencing, but it doesn't seem to be interfering with any functionality.

The warning should be solved with the latest commit.

**Review by MarcMData — DISMISSED** [review]
Tested locally on a Pixel C API 31 emulator. Upgraded flutter as per the guide in the PR without issue. Had to install additional dependencies in Visual Studio, as advised from running Flutter Doctor.

Following the requested changes everything now runs as it should, without warnings. Approved from here.

**Review by NikolajKrogh — DISMISSED** [review]
lgtm

**Review by NikolajKrogh — APPROVED** [review]

**Review by MarcMData — APPROVED** [review]

**Pull request merged**